Created initial classes and methods.

paulaOliverosHollis opened the pull request:
Problems - this class will hold all of the problems as methods of their own.

DynamicProblemPrinter - By using reflection, we can call any problem from the problems class which in turn will be shown to the screen.

Program- I added some of the methods that will be use to interact with the user.

Problem1()- Created the method and solve the problem. I tested the problem on LinqPad5 and it works perfectly.

paulaOliverosHollis committed:
Created initial classes and methods.

Problems - this class will hold all of the problems as methods of their own.

DynamicProblemPrinter - By using reflection, we can call any problem from the problems class which in turn will be shown to the screen.

Program- I added some of the methods that will be use to interact with the user.

zhollis21 commented on Project-Euler/Project-Euler/DynamicProblemPrinter.cs:
```diff
@@ -0,0 +1,26 @@
+
+namespace Project_Euler
+{
+    class DynamicProblemPrinter
```
This should be a public static class because it doesn't need to know anything about itself to do it's job or need to store anything

zhollis21 replied:
Also, this is more of a problem runner not a problem printer

paulaOliverosHollis replied:
Those are very good points. I have already fixed both issues and committed.

zhollis21 commented on Project-Euler/Project-Euler/DynamicProblemPrinter.cs:
```diff
@@ -0,0 +1,26 @@
+        {
+            var method = objectType.GetMethod($"Problem{problemNumber}");
+
+            method.Invoke(null, null);
```
I would wrap this in a if method isn't null block, because if anyone enters in a number (like 6) that you don't have this will crash the app

paulaOliverosHollis replied:
Thank you for pointing that out, I agree. You can see this change in my latest commit.

zhollis21 commented on Project-Euler/Project-Euler/DynamicProblemPrinter.cs:
```diff
@@ -0,0 +1,26 @@
+        {
+            for (int i = 1; i < 26; i++)
+            {
+                PrintProblem(i, typeof(Problems));
```
This will crash the app unless you fix the issue above

paulaOliverosHollis replied:
Fixed!

zhollis21 commented on Project-Euler/Project-Euler/DynamicProblemPrinter.cs:
```diff
@@ -0,0 +1,26 @@
+        {
+            var method = objectType.GetMethod($"Problem{problemNumber}");
+
+            method.Invoke(null, null);
```
It might also be nice to wrap this in a timer so the user knows how long each method took. i.e.

Record current time
Invoke Method
Print current time - start time

paulaOliverosHollis replied:
I like that idea. However, I will add this later on once I have added more problems.

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -0,0 +1,34 @@
+{
+    public static class Problems
+    {
+        private static void Problem1()
```
Nice job!

zhollis21 replied:
Just kidding, this will have to be public or you can't call it

paulaOliverosHollis replied:
True, Ok.

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -8,8 +8,31 @@ namespace Project_Euler
 {
     class Program
```
public

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -8,8 +8,31 @@ namespace Project_Euler
+            PrintMenuOptions();//Eventually this method will not be called in main directly. This is for testing purposes.
+        }
+
+        public static void PrintIntro()
```
private

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -8,8 +8,31 @@ namespace Project_Euler
+        {
+            Console.WriteLine("\n\n\tSelect One Of The Following Options:\n" +
+                              "\n\tA - See All Problems and Solutions" +
+                              "\n\tB - See One Specific Problem and Solution" +
```
Why B?

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -8,8 +8,31 @@ namespace Project_Euler
+                              "\n\tB - See One Specific Problem and Solution" +
+                              "\n\tQ - Quit");
+
+            Console.ReadLine();
```
You shouldn't be reading input in a method called PrintMenu, I would create another method which gets the user input

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -0,0 +1,34 @@
+            Console.WriteLine($"Answer: {sum}");
+        }
+
+        private static void Problem2()
```
You can't call a private method from your DynamicProblemPrinter class, make it public

paulaOliverosHollis committed:
Fixed small issues in DynamicProblemPrinter.

Changed the name of the class to DynamicProblemSelector since this name is more appropriate.

The class is now public static since it is the most logical approach.

Wrapped the method.Invoke in an if statement that checks for null to avoid any disasters in case user enters a problem number we do not have or is simply incorrect.

paulaOliverosHollis commented on Project-Euler/Project-Euler/DynamicProblemSelector.cs:
```diff
@@ -0,0 +1,33 @@
+            }
+            else
+            {
+                Console.WriteLine($"Problem {problemNumber} is not a valid selection. Please try again!");
```
Very Nice!

paulaOliverosHollis commented:
Since I need to merge this branch to be able to see the code on my computer, I am going to go ahead and merge it now but I will immediately change what you requested and commit it. I will let you know when I do so.
